File: DAL/Models/CompanyType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.Models
{
    public class CompanyType
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public ICollection<Company> Companies { get; set; }
    }
}

File: GlassDoor/ClientApp/src/app/models/company-type.ts
import { Company } from "./company";

export class CompanyType {

  constructor(id?: number, name?: string, companies?: Company) {

    this.id = id;
    this.name = name;
    this.companies = companies;

  }
  public id: number;
  public name: string;
  public companies: Company;
}
